compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2331,6 +2331,18 @@ public void visit(BLangMatchClause matchClause) {
             SymbolEnv patternEnv = SymbolEnv.createPatternEnv(matchPattern, env);
             analyzeNode(matchPattern, patternEnv);
             resolveMatchClauseVariableTypes(matchPattern, clauseVariables, blockEnv);
+
+            if (matchPattern.getKind() == NodeKind.VAR_BINDING_PATTERN_MATCH_PATTERN &&
+                    (matchPattern.matchExpr != null ? matchPattern.matchExpr.getKind() : null)
```

Comment:
When can `matchPattern.matchExpr` be null?

Reply:
Yes. we dont need that, removed